test/integration/user_management_test.rb

```diff
@@ -0,0 +1,149 @@
+require 'test_helper'
+
+class UserManagementTest < ActionDispatch::IntegrationTest
+
+  before do
+    Capybara.current_driver = :poltergeist
+    @user = create :user,
+                    name: 'john',
+                    email: 'user@dcaf.com',
+                    role: :cm
+
+    @admin = create :user,
+                    name: 'admin',
+                    email: 'admin@dcaf.com',
+                    role: :admin
+
+    @data_volunteer = create :user,
+                            name: 'volunteer',
+                            email: 'data@gmail.com',
+                            role: :data_volunteer
+
+  end
+
+  describe 'case manager' do
+    before do
+      log_in @user
+    end
+
+    it 'should not be able to access Admin links' do
+      assert has_no_link? 'Admin'
+    end
+  end
+
+  describe 'data volunteer' do
+    before do
+      log_in @data_volunteer
+    end
+
+    it 'should not be able to access user management' do
+      click_link 'Admin'
+      assert_no_text 'User Management'
+    end
+
+  end
+
+  describe 'admin user' do
+    before do
+      log_in @admin
+    end
+
+    it 'should be able to access user management' do
+      click_link 'Admin'
+      assert_text 'User Management'
+      click_link 'User Management'
+      wait_for_element 'User Account Management'
+      click_link 'Add New User'
```

Comment:
Can you throw some `assert has_content? 'Whatever text'` checks in here, so it yields more helpful error messages if it ever chokes?